src/plugins/dashboard_embeddable_container/public/plugin.tsx

```diff
@@ -50,10 +63,21 @@ export class DashboardEmbeddableContainerPublicPlugin
   implements Plugin<Setup, Start, SetupDependencies, StartDependencies> {
   constructor(initializerContext: PluginInitializerContext) {}
 
-  public setup(core: CoreSetup, { embeddable, uiActions }: SetupDependencies): Setup {
+  public setup(
+    core: CoreSetup,
+    { directAccessLinks, embeddable, uiActions }: SetupDependencies
+  ): Setup {
     const expandPanelAction = new ExpandPanelAction();
     uiActions.registerAction(expandPanelAction);
     uiActions.attachAction(CONTEXT_MENU_TRIGGER, expandPanelAction.id);
+    const startServices = core.getStartServices();
+    directAccessLinks.registerAccessLinkGenerator(
+      DASHBOARD_APP_LINK_GENERATOR,
+      createDirectAccessDashboardLinkGenerator(async () => ({
+        appBasePath: (await startServices)[0].application.getUrlForApp('dashboard'),
+        useHashedUrl: (await startServices)[0].uiSettings.get('state:storeInSessionStorage'),
+      }))
+    );
```

Comment:
This feels like it should be in the `dashboard` plugin not in `dashboard_embeddable_container` plugin. I thought the idea of Dashboard Embeddable Container that we potentially can render it anywhere. Also it is the concern of Dashboard app to know about its state, the `_a` and `_g` keys and `setStateToKbnUrl` function, not container's.

Reply:
My thought was to rename this plugin, make it `dashboard`, and contain the embeddable as a service.